samza-api/src/main/java/org/apache/samza/metrics/ListGaugeEvictionPolicy.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.metrics;
+
+public interface ListGaugeEvictionPolicy<T> {
```

Comment:
Prefer to rethink this interface as follows: 
```
public class Evictor {
  public void evict(List<T> elements);
}
```
Evictors can mutate the list they are passed. In addition to getting rid off the setter, you should be able to support arbitrary eviction policies

Reply:
talked offline